File: 1457-pseudo-palindromic-paths-in-a-binary-tree/1457-pseudo-palindromic-paths-in-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int dfs(TreeNode* root,unordered_map<int,int>&mp,int &ans)
    {
        if(root==NULL)
        {
            return 1;
        }
        mp[root->val]++;
        int a= dfs(root->left,mp,ans);
        int b=dfs(root->right,mp,ans);
        if(a&b)
        {
            int flag=0;
            for(auto it:mp)
            {
                if(it.second%2)flag++;
            }
            if(flag<2)
                ans++;
        }
        mp[root->val]--;
        return 0;
        
    }
    int pseudoPalindromicPaths (TreeNode* root) {
        unordered_map<int,int>mp;
        int ans=0;
        dfs(root,mp,ans);
        return ans;
        
    }
};

File: 1457-pseudo-palindromic-paths-in-a-binary-tree/1457-pseudo-palindromic-paths-in-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int recur(TreeNode root,int []dp){
        int flag=0;
        int x=0;
        dp[root.val]++;
        if(root.left==null){
            flag++;
        }else{
            x+=recur(root.left,dp);
        }
        if(root.right==null){
            flag++;
        }else{
            x=x+recur(root.right,dp);
        }
        if(flag==2){
            int check=0;
            for(int i=0;i<dp.length;i++){
                if(dp[i]%2==1){
                    check++;
                }
            }
            if(check<2)
                x++;
        }
        dp[root.val]--;
        return x;
    }
    public int pseudoPalindromicPaths (TreeNode root) {
        
        int []dp=new int[10];
        return recur(root,dp);
    }
}